src/KM_BuildList.pas

```diff
@@ -782,19 +786,28 @@ procedure TKMHousePlanList.RemPlan(aLoc: TKMPoint);
 end;
 
 
-function TKMHousePlanList.GetPlan(aLoc: TKMPoint): THouseType;
+function TKMHousePlanList.GetEmptyPlan: TKMHousePlan;
+begin
+  Result.HouseType := ht_None;
+  Result.Loc := ZERO_POINT;
+  Result.JobStatus := js_Empty;
+  Result.Worker := nil;
+end;
+
+
+function TKMHousePlanList.GetPlan(aLoc: TKMPoint): TKMHousePlan;
 var
   I: Integer;
 begin
-  Result := ht_None;
+  Result := GetEmptyPlan;
```

Comment:
I would prefer the function to return a boolean for success and pass the plan via an out parameter rather than having GetEmptyPlan.